### PR TITLE
Deprecate/utils boto3 elasticsearch

### PR DESCRIPTION
**Pull request by major0**
### What does this PR do?

Deprecate usage of `utils/boto3_elasticsearch.py`

### Previous Behavior
The `utils/boto3_elasticsearch.py` was a thin wrapper around botocore to supply the boto3_elasticsearch with the ability to query the _waiter_ objects.  Presumably this was due to a bug in Boto3 at one point in which the waiters where not made available in the ES connection object.

> Botocore waiters for elasticsearch that are not present in boto3+botocore (yet).

The methods in question have been available in the ES connection object for some time now and these utility functions are no longer necessary.

### New Behavior
Use the ES `conn.get_waiter()` from the ES object instead of the botocore.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by sagetherage** [comment]
@major0 we are working on these test failures - they likely don't have to do with your PR

File: salt/utils/boto3_elasticsearch.py [hunk 1]
```diff
@@ -99,6 +94,7 @@ def __virtual__():
     return salt.utils.versions.check_boto_reqs(check_boto=False)
 
 
+@salt.utils.decorators.is_deprecated(globals(), "Phosphorus")
```
Comment (Ch3LL) [inline salt/utils/boto3_elasticsearch.py:97]:
```suggestion
@salt.utils.decorators.is_deprecated(globals(), "Sulfur")
```

File: salt/utils/boto3_elasticsearch.py [hunk 2]
```diff
@@ -124,6 +120,7 @@ def get_waiter(client, waiter=None, waiter_config=None):
     return botocore.waiter.create_waiter_with_client(waiter, waiter_model, client)
 
 
+@salt.utils.decorators.is_deprecated(globals(), "Phosphorus")
```
Comment (Ch3LL) [inline salt/utils/boto3_elasticsearch.py:123]:
```suggestion
@salt.utils.decorators.is_deprecated(globals(), "Sulfur")
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]
Just one small change and then we can get this merged.

**Review by Ch3LL — APPROVED** [review]

**Review by waynew — APPROVED** [review]

**Pull request merged**